lib/Sema/CSSimplify.cpp

```diff
@@ -2060,17 +2060,13 @@ static bool isInPatternMatchingContext(ConstraintLocatorBuilder locator) {
 
   if (!path.empty()) {
     // Direct pattern matching between tuple pattern and tuple type.
-    if (path.back().is<LocatorPathElt::PatternMatch>()) {
-      return true;
-    } else if (path.size() > 1) {
-      // sub-pattern matching as part of the enum element matching
-      // where sub-element is a tuple pattern e.g.
-      // `case .foo((a: 42, _)) = question`
-      auto lastIdx = path.size() - 1;
-      if (path[lastIdx - 1].is<LocatorPathElt::PatternMatch>() &&
-          path[lastIdx].is<LocatorPathElt::FunctionArgument>())
-        return true;
-    }
+    // sub-pattern matching as part of the enum element matching
+    // where sub-element is a tuple pattern e.g.
+    // `case .foo((a: 42, _)) = question`
+    auto pathElement = llvm::find_if(path, [](LocatorPathElt &elt) {
```

Comment:
There is a better llvm function for this - llvm::is_contained which returns a bool

Reply:
sorry to reply late. tried to fix it, but it looks redundant than current code as below.

```
std::vector<ConstraintLocator::PathElementKind> kinds;
llvm::transform(path, std::back_inserter(kinds), [](LocatorPathElt &elt) {
    return elt.getKind();
});
return llvm::is_contained(kinds, ConstraintLocator::PatternMatch);

```
Could you please let me know if you have any other good code?

Reply:
Ah `PatternMatch` requires a `pattern` argument, so attempting to match it via llvm::is_contained wouldn't work :/ Okay, let's keep it as `find_if` then.